sql/expression/function/greatest_least.go

```diff
@@ -0,0 +1,329 @@
+package function
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// Greatest returns the argument with the greatest numerical or string value. It allows for
+// numeric (ints anf floats) and string arguments and will return the used type
+// when all arguments are of the same type or floats if there are numerically
+// convertible strings or integers mixed with floats. When ints or floats
+// are mixed with non numerically convertible strings, those are ignored.
+type Greatest struct {
+	args []sql.Expression
+}
+
+// ErrUnsupportedType is returned when an argument to Greatest or Latest is not numeric or string
+var ErrUnsupportedType = errors.NewKind("unsupported type for greatest/latest argument: %T")
+
+// NewGreatest creates a new Greatest UDF
+func NewGreatest(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) == 0 {
+		return nil, sql.ErrInvalidArgumentNumber.New("GREATEST", "1 or more", 0)
+	}
+
+	for _, arg := range args {
+		if !arg.Resolved() {
```

Comment:
When it's first parsed, nothing that's not a literal will be resolved, so they will be ignored. So this if should be removed.

Reply:
Fixed in next push.